bpf/lib/common.h

```diff
@@ -726,6 +726,7 @@ enum {
 	SVC_FLAG_LOADBALANCER = (1 << 5),  /* LoadBalancer service */
 	SVC_FLAG_ROUTABLE     = (1 << 6),  /* Not a surrogate/ClusterIP entry */
 	SVC_FLAG_SOURCE_RANGE = (1 << 7),  /* Check LoadBalancer source range */
+	SVC_FLAG_INT_LOCAL_SCOPE = (1 << 12), /* internalTrafficPolicy=Local */
```

Comment:
nit: Could you extend the commit message on how the result looks like with a simple service in k8s with combinations of for example:

- internalTrafficPolicy={Local,Cluster}
- externalTrafficPolicy={Local,Cluster}
- internalTrafficPolicy={Local,Cluster} and externalTrafficPolicy={Local,Cluster}

Reply:
In terms of documentation we do have a section in here: https://docs.cilium.io/en/latest/network/kubernetes/kubeproxy-free/#client-source-ip-preservation (Documentation/network/kubernetes/kubeproxy-free.rst), but this is mainly user facing. I think we should definitely extend the documentation on this as well that we support this too. In terms of implementation details iow how the svc/bpf map gets populated, we don't have a concrete document hence my thinking that this could perhaps go as part of the commit message. Other option could be to add a paragraph into the doc explaining the behavior with svc list examples and what these entries do.

Reply:
> Maybe it's better to put the table from the commit message somewhere in the documentation?

This is about the table in patch 2? I feels that level of detail actually works well in the commit description - it reflects the current implementation, and is the natural point where developers will go when trying to understand it.